Unknown CMake command "get_target_output_dir" in BareosSystemtestFunctions.cmake line 122

### Bareos component version

24.0.2

### Steps to reproduce

rpmbuild -ba -v bareos.spec

### Expected results

RPM packages get build

### Actual results

```
...
Entering /opt/dev/bareos/rpmbuild/20250327-132831/BUILD/bareos/systemtests
   SYSTEMTESTS_S3_USE_HTTPS: ON
-- POSTGRES_BIN_PATH is /usr/bin
-- RUN_SYSTEMTESTS_ON_INSTALLED_FILES="OFF"
-- Looking for binaries and paths...
CMake Error at systemtests/cmake/BareosSystemtestFunctions.cmake:122 (get_target_output_dir):
  Unknown CMake command "get_target_output_dir".
Call Stack (most recent call first):
  systemtests/cmake/BareosSystemtestFunctions.cmake:190 (find_compiled_binary_and_set_binary_name_to_test_variable_for)
  systemtests/CMakeLists.txt:181 (find_systemtests_binary_paths)

```

### Environment

```markdown
Linux (various distributions an versions)
```

### Relevant log output

```shell

```

### Relevant traces output

```shell

```

### Anything else?

Patch suggestion:

```
--- bareos/systemtests/CMakeLists.txt	2025-03-27 10:03:40.000000000 +0100
+++ bareos/systemtests/CMakeLists.txt	2025-03-27 13:09:22.014428254 +0100
@@ -85,6 +85,7 @@
 endif()
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+include(BareosTargetTools)
 include(BareosSystemtestFunctions)
 
 # ps is required for systemtests

```

Sure, observed on Alma Linux 8.10 and Rocky Linux 9.5 both with cmake 3.26.5.

I can reproduce with the following script, which is a stripped down version of the one I am using:

```
#!/bin/bash

CENTOS_VERSION=900

GITHUB_REPOS=https://github.com/bareos/bareos
SRC_FILE=24.0.2.tar.gz
SRC_URL=${GITHUB_REPOS}/archive/Release/$SRC_FILE
SRC_DIR=bareos-Release-24.0.2
SPEC_PATH=$SRC_DIR/core/platforms/packaging/bareos.spec

RPM_BUILD=`pwd`/rpmbuild/

mkdir -p ${RPM_BUILD}/SPECS ${RPM_BUILD}/SOURCES

echo "Changing to ${RPM_BUILD}/SOURCES ..."
cd ${RPM_BUILD}/SOURCES
echo "Downloading ${SRC_URL} ..."
wget ${SRC_URL}
echo "Extracting ${SPEC_PATH} from tarball ..."
tar xfz 24.0.2.tar.gz ${SPEC_PATH}
if [ ! -f ${SPEC_PATH} ]
then
    echo "Could not find spec file at '${SPEC_PATH}' - fix script - Aborting!"
    exit 1
fi
mv ${SPEC_PATH} ../SPECS/
rm -rf ${SRC_DIR}

echo "Changing to ../SPECS ..."
cd ../SPECS

cp -p bareos.spec bareos.spec.saved
echo "Adjusting bareos.spec ..."
sed -i \
    -e '/^Version:/ s#0.0.0#'"24.0.2"'#g' \
    -e '/^%define  *droplet  *1$/ s#1#0#g' \
    -e '/^%define  *glusterfs  *1$/ s#1#0#g' \
    -e '/^%define  *ceph  *1$/ s#1#0#g' \
    -e '/^%define  *enable_grpc  *1$/ s#1#0#g' \
    -e '/^Source0:/ s# .*# %{version}.tar.gz#' \
    bareos.spec
echo "Diff for bareos.spec:"
diff -u bareos.spec.saved bareos.spec

echo "Calling rpmbuild with topdir ${RPM_BUILD} and centos_version ${CENTOS_VERSION} ..."
rpmbuild \
    --define="_topdir ${RPM_BUILD}" \
    --define="centos_version ${CENTOS_VERSION}" \
    -ba -v bareos.spec \
    > rpmbuild-24.0.2.out 2>&1
echo "Check rpmbuild output in `pwd`/rpmbuild-24.0.2.out"

```
The script output is:

```
Changing to /opt/dev/bareos/tmp/rpmbuild//SOURCES ...
Downloading https://github.com/bareos/bareos/archive/Release/24.0.2.tar.gz ...
--2025-03-27 16:34:39--  https://github.com/bareos/bareos/archive/Release/24.0.2.tar.gz
Resolving github.com (github.com)... 140.82.121.3
Connecting to github.com (github.com)|140.82.121.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/bareos/bareos/tar.gz/refs/tags/Release/24.0.2 [following]
--2025-03-27 16:34:40--  https://codeload.github.com/bareos/bareos/tar.gz/refs/tags/Release/24.0.2
Resolving codeload.github.com (codeload.github.com)... 140.82.121.10
Connecting to codeload.github.com (codeload.github.com)|140.82.121.10|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/x-gzip]
Saving to: ‘24.0.2.tar.gz’

24.0.2.tar.gz                            [       <=>                                                             ]  12.73M  9.91MB/s    in 1.3s    

2025-03-27 16:34:41 (9.91 MB/s) - ‘24.0.2.tar.gz’ saved [13350863]

Extracting bareos-Release-24.0.2/core/platforms/packaging/bareos.spec from tarball ...
Changing to ../SPECS ...
Adjusting bareos.spec ...
Diff for bareos.spec:
--- bareos.spec.saved	2025-03-27 10:03:40.000000000 +0100
+++ bareos.spec	2025-03-27 16:34:42.481072687 +0100
@@ -5,7 +5,7 @@
 #
 
 Name:       bareos
-Version:    0.0.0
+Version:    24.0.2
 Release:    0
 Group:      Productivity/Archiving/Backup
 License:    AGPL-3.0
@@ -38,14 +38,14 @@
 %define client_only 0
 %define build_qt_monitor 1
 %define glusterfs 0
-%define droplet 1
+%define droplet 0
 %define have_git 1
 %define install_suse_fw 0
 %define systemd_support 0
 %define python_plugins 1
 %define contrib 1
 %define webui 1
-%define enable_grpc 1
+%define enable_grpc 0
 
 # cmake build directory
 %define CMAKE_BUILDDIR       cmake-build
@@ -91,7 +91,7 @@
 #
 
 %if 0%{?fedora} >= 20
-%define glusterfs 1
+%define glusterfs 0
 %define systemd_support 1
 %endif
 
@@ -100,7 +100,7 @@
 %endif
 
 %if 0%{?rhel} >= 7 && (0%{?rhel} <= 9)
-%define glusterfs 1
+%define glusterfs 0
 %endif
 
 %if 0%{?rhel} == 7
@@ -146,7 +146,7 @@
 BuildRequires: git-core
 %endif
 
-Source0: %{name}-%{version}.tar.gz
+Source0: %{version}.tar.gz
 
 BuildRequires: cmake >= 3.17
 BuildRequires: gcc
Calling rpmbuild with topdir /opt/dev/bareos/tmp/rpmbuild/ and centos_version 900 ...
Check rpmbuild output in /opt/dev/bareos/tmp/rpmbuild/SPECS/rpmbuild-24.0.2.out

```

and the actual rpmbuild output showing the cmake ends with:

```
Entering /opt/dev/bareos/tmp/rpmbuild/BUILD/bareos/systemtests
   SYSTEMTESTS_S3_USE_HTTPS: ON
-- POSTGRES_BIN_PATH is /usr/bin
-- RUN_SYSTEMTESTS_ON_INSTALLED_FILES="OFF"
-- Looking for binaries and paths...
CMake Error at systemtests/cmake/BareosSystemtestFunctions.cmake:122 (get_target_output_dir):
  Unknown CMake command "get_target_output_dir".
Call Stack (most recent call first):
  systemtests/cmake/BareosSystemtestFunctions.cmake:190 (find_compiled_binary_and_set_binary_name_to_test_variable_for)
  systemtests/CMakeLists.txt:181 (find_systemtests_binary_paths)


-- Configuring incomplete, errors occurred!
error: Bad exit status from /var/tmp/rpm-tmp.8bMQHk (%build)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.8bMQHk (%build)

```

I will try to attach the full rpmbuild output (shortened by 12000 lines of verbose tar xf output).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -27,6 +27,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - config: fix issues with config directive aliases [PR #2159]
 - cleanup: fix multi-config tests [PR #2202]
 - mssqlvdi-fd: add support for filestream backups [PR #2072]
+- cmake: fix not including BareosTargetTools in systemtests, needed for get_target_output_dir [PR #2232]
 
 ### Documentation
 - docs: fix grpc-fd plugin call [PR #2068]
@@ -110,4 +111,5 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 [PR #2202]: https://github.com/bareos/bareos/pull/2202
 [PR #2211]: https://github.com/bareos/bareos/pull/2211
 [PR #2220]: https://github.com/bareos/bareos/pull/2220
+[PR #2232]: https://github.com/bareos/bareos/pull/2232
 [unreleased]: https://github.com/bareos/bareos/tree/master

--- a/systemtests/CMakeLists.txt
+++ b/systemtests/CMakeLists.txt
@@ -85,6 +85,7 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
 endif()
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+include(BareosTargetTools)
 include(BareosSystemtestFunctions)
 
 # ps is required for systemtests

